src/OrbitDB.js

```diff
@@ -59,19 +74,32 @@ let databaseTypes = {
 
     const { id } = await ipfs.id()
     const directory = options.directory || './orbitdb'
-    const keystore = options.keystore || Keystore.create(path.join(directory, id, '/keystore'))
 
-    const identity = options.identity || await Identities.createIdentity({
-      id: options.id || id,
-      keystore: keystore,
-    })
-    options = Object.assign({}, options, {
+    if(!options.keystore) {
+      let keyStorage = await storage.createStore(path.join(directory, id, '/keystore'))
+      options.keystore = new Keystore(keyStorage)
+    }
+
+    if(!options.identity) {
+      options.identity = await Identities.createIdentity({
+        id: options.id || id,
+        keystore: options.keystore,
+      })
+    }
+
+    if(!options.cache) {
+      let cacheStorage = await storage.createStore(path.join(directory, id, '/cache'))
+      options.cache = new Cache(cacheStorage)
+    }
+
+    const finalOptions = Object.assign({}, options, {
       peerId: id ,
       directory: directory,
-      keystore: keystore
+      keystore: options.keystore,
+      cache: options.cache
```

Comment:
Are these two in fact needed? It seems to me we already assign new values to `options.cache` and `.keystore` above.